core/src/main/java/me/leoko/advancedban/utils/Command.java

```diff
@@ -291,9 +293,20 @@ else if(args.length == 2)
                 else
                     return list();
             }),
-            new ListProcessor(
-                    target -> PunishmentManager.get().getPunishments(target, null, false),
-                    "History", true, true),
+            input -> {
+                MethodInterface mi = Universal.get().getMethods();
+                List<PunishmentType> putList = new ArrayList<>();
+                mi.getStringList(mi.getConfig(),"FullHistory").forEach((typeString -> putList.add(PunishmentType.valueOf(typeString))));
+
+                if (!Universal.get().hasPerms(input.getSender(), "ab.history")) {
+                    MessageManager.sendMessage(input.getSender(), "General.NoPerms", true);
+                    return;
+                }
```

Comment:
I don't think that you need to perform a permission check here as the permission for the command is specified a few lines above